app/static/js/report_map.js

```diff
@@ -7,22 +8,63 @@ $(document).ready(function () {
 });
 
 function grabMyPosition() {
-    if (navigator.geolocation) {
-        return navigator.geolocation.getCurrentPosition(centerMap);
+    var geoOptions = { maximumAge: 30000,  //  Valid for 5 minutes
+        timeout:5000,  // Wait 5 seconds
+        enableHighAccuracy:true
+    }
+
+    // If there's geolocation, try to get user coords.
+    if(navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(success, error, geoOptions);
     } else {
-        //If geolocation doesn't work, the map defaults to San Jose.
+    //If geolocation isn't available, the map defaults to San Jose.
         var lat = 37.3394444;
         var lon = -121.8938889;
```

Comment:
These numbers are hard-coded in multiple places. They should be defined in one place.